src/statement/statement.ts

```diff
@@ -65,13 +71,21 @@ export class Statement {
     return errors;
   }
 
+  validateForNotResourcePolicy() {
+    const errors = this.validateForAnyPolicy();
+    if (Object.keys(this.principals).length === 0) {
+      errors.push(`Statement(${this.sid}) must specify at least one IAM principal.`);
+    }
+    return errors;
+  }
+
```

Comment:
Why was this method added? I think this is not necessary.

`validateForResourcePolicy` validates a statement in the context of a Policy attached to an AWS Resource like an S3 bucket, a KMS Key, a SecretsManager secret, ...
Likewise `validateForIdentityPolicy` validates a statement in the context of a policy attached to an AWS Identity like an IAM User, Group, or Role.

The `ForResource` has nothing to do with `statement.resources`. Therefore no need to add a `ForNotResource` variant.

Reply:
Got it. Removed the method.